### PR TITLE
Print branch name and the latest commit of the current execution session

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
The test summary also shows what branch and the latest commit were.

<Test result>

[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 08:42:44
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:5

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 2.77
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS

Logs can be found at C:\LISAv2\WI48\TestResults\2019-25-01-00-42-40-6857

**01/25/2019 08:48:00 : [INFO ] ----------------------------------------------------------
01/25/2019 08:48:00 : [INFO ] Current Branch:  juhle/issue652
01/25/2019 08:48:00 : [INFO ] Currnet Commit ID: 2cb4e369591cc145ebf970c6c1d7ffb6b1d06aab
01/25/2019 08:48:00 : [INFO ] ----------------------------------------------------------**
01/25/2019 08:48:00 : [INFO ] Creating 'C:\LISAv2\WI48\Azure-WI48-TestLogs.zip' from 'C:\LISAv2\WI48\TestResults\2019-25-01-00-42-40-6857'
01/25/2019 08:48:00 : [INFO ] C:\LISAv2\WI48\Azure-WI48-TestLogs.zip created successfully.
01/25/2019 08:48:00 : [INFO ] Analyzing results..
01/25/2019 08:48:00 : [INFO ] PASS  : 1
01/25/2019 08:48:00 : [INFO ] FAIL  : 0
01/25/2019 08:48:00 : [INFO ] ABORT : 0
01/25/2019 08:48:00 : [INFO ] Copying all files back to original working directory: C:\Users\juhlee\source\repo\LISAv2.
01/25/2019 08:48:04 : [INFO ] Cleaning up C:\LISAv2\WI48
01/25/2019 08:48:04 : [INFO ] Setting workspace back to original location: C:\Users\juhlee\source\repo\LISAv2
01/25/2019 08:48:04 : [INFO ] LISAv2 exit code: 0
PS C:\Users\juhlee\source\repo\LISAv2>

### CODE REVIEW

**Comment by chvalean** [comment]
Some scenarios not considered here:
1. lisav2 can be taken as an archive, so there is no git branch to show.
2. if running from jenkins, this information is displayed in the begging where jenkins handles the SCM.

File: LISAv2-Framework.psm1 [hunk 1]
```diff
@@ -78,6 +78,15 @@ function Start-LISAv2 {
 			Set-Variable -Name "TestId" -Value $testId -Scope Global -Force
 			Write-LogInfo "Autogenerated test ID: $testId"
 
+			# Get git information
+			git branch | foreach {
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: I would not make git a dependency of lisav2, what if the lisav2 code has been downloaded from the releases page or using zip download from github?

File: LISAv2-Framework.psm1 [hunk 2]
```diff
@@ -167,6 +176,11 @@ function Start-LISAv2 {
 			$plainTextSummary = $testController.TestSummary.GetPlainTextSummary()
 			Write-LogInfo $plainTextSummary
 
+			Write-LogInfo "------------------------------------------------------------"
+			Write-LogInfo "Current Branch: $current_branch"
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: what if you are using a tag, not branch?

File: LISAv2-Framework.psm1 [hunk 3]
```diff
@@ -167,6 +176,11 @@ function Start-LISAv2 {
...
+			Write-LogInfo "Currnet Commit ID: $CommitId"
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: typo

File: LISAv2-Framework.psm1 (hunk 2 above)
Comment (juhlee-microsoft) [reply LISAv2-Framework.psm1]: We don't use tag yet. It's good point but won't apply that situation to current change.

File: LISAv2-Framework.psm1 (hunk 1 above)
Comment (juhlee-microsoft) [reply LISAv2-Framework.psm1]: Fixed and tested in local downloaded folder. Below is the test screenshot. Using try-catch-finally statement.
<Inside Git repo>
PS C:\Users\juhlee\Source\Repos\LISAv2> .\Untitled3.ps1
hello starting
taken care of it:  juhle/issue652
I am out of try-catch-finally
really done:  juhle/issue652

<No Git repo>
PS C:\Users\juhlee\Source\Repos\LISAv2> cd C:\Users\juhlee\Downloads\LISAv2-master\LISAv2-master

PS C:\Users\juhlee\Downloads\LISAv2-master\LISAv2-master> .\Untitled3.ps1
hello starting
git : fatal: not a git repository (or any of the parent directories): .git
At C:\Users\juhlee\Downloads\LISAv2-master\LISAv2-master\Untitled3.ps1:5 char:9
+         git branch | foreach {
+         ~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (fatal: not a gi...ectories): .git:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
taken care of it: Not a git repo
I am out of try-catch-finally
really done: Not a git repo

**Comment by juhlee-microsoft** [comment]
> Some scenarios not considered here:
> 
> 1. lisav2 can be taken as an archive, so there is no git branch to show.
> 2. if running from jenkins, this information is displayed in the begging where jenkins handles the SCM.

@chvalean Fixed in 3f14198582aab626aa1a5a237643058a3825bd21

Will test in Jenkins and post the result soon.

File: LISAv2-Framework.psm1 (hunk 3 above)
Comment (juhlee-microsoft) [reply LISAv2-Framework.psm1]: fixed in 3f14198582aab626aa1a5a237643058a3825bd21

File: LISAv2-Framework.psm1 (hunk 1 above)
Comment (juhlee-microsoft) [reply LISAv2-Framework.psm1]: fixed in 3f14198582aab626aa1a5a237643058a3825bd21

**Comment by juhlee-microsoft** [comment]
Test Result comparison between git repo and none.


[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 18:12:45
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:5

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 2.93
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS


Logs can be found at C:\LISAv2\BL24\TestResults\2019-25-01-10-12-42-1950


01/25/2019 18:18:05 : [INFO ] ----------------------------------------------------------
01/25/2019 18:18:05 : [INFO ] Current Branch:  juhle/issue652
01/25/2019 18:18:05 : [INFO ] Currnet Commit ID: 3f14198582aab626aa1a5a237643058a3825bd21
01/25/2019 18:18:05 : [INFO ] ----------------------------------------------------------
01/25/2019 18:18:05 : [INFO ] Creating 'C:\LISAv2\BL24\Azure-BL24-TestLogs.zip' from 'C:\LISAv2\BL24\TestResults\2019-25-01-10-12-42-1950'
01/25/2019 18:18:05 : [INFO ] C:\LISAv2\BL24\Azure-BL24-TestLogs.zip created successfully.
01/25/2019 18:18:05 : [INFO ] Analyzing results..
01/25/2019 18:18:05 : [INFO ] PASS  : 1
01/25/2019 18:18:05 : [INFO ] FAIL  : 0
01/25/2019 18:18:05 : [INFO ] ABORT : 0
01/25/2019 18:18:05 : [INFO ] Copying all files back to original working directory: C:\Users\juhlee\source\repo\LISAv2.
01/25/2019 18:18:09 : [INFO ] Cleaning up C:\LISAv2\BL24
01/25/2019 18:18:09 : [INFO ] Setting workspace back to original location: C:\Users\juhlee\source\repo\LISAv2
01/25/2019 18:18:09 : [INFO ] LISAv2 exit code: 0
PS C:\Users\juhlee\source\repo\LISAv2> 


[LISAv2 Test Results Summary]
Test Run On           : 01/25/2019 18:46:33
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:6

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 4.28
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS


Logs can be found at C:\LISAv2\QC84\TestResults\2019-25-01-10-46-28-5927


01/25/2019 18:53:06 : [INFO ] ------------------------------------------------------------
01/25/2019 18:53:06 : [INFO ] Current Branch: Not git repo
01/25/2019 18:53:06 : [INFO ] Current Commit ID: 0
01/25/2019 18:53:06 : [INFO ] ------------------------------------------------------------
01/25/2019 18:53:06 : [INFO ] Creating 'C:\LISAv2\QC84\Azure-QC84-TestLogs.zip' from 'C:\LISAv2\QC84\TestResults\2019-25-01-10-46-28-5927'
01/25/2019 18:53:06 : [INFO ] C:\LISAv2\QC84\Azure-QC84-TestLogs.zip created successfully.
01/25/2019 18:53:06 : [INFO ] Analyzing results..
01/25/2019 18:53:06 : [INFO ] PASS  : 1
01/25/2019 18:53:06 : [INFO ] FAIL  : 0
01/25/2019 18:53:06 : [INFO ] ABORT : 0
01/25/2019 18:53:06 : [INFO ] Copying all files back to original working directory: C:\Users\juhlee\Downloads\LISAv2-juhle-issue652\LISAv2-juhle-issue652.
01/25/2019 18:53:10 : [INFO ] Cleaning up C:\LISAv2\QC84
01/25/2019 18:53:10 : [INFO ] Setting workspace back to original location: C:\Users\juhlee\Downloads\LISAv2-juhle-issue652\LISAv2-juhle-issue652
01/25/2019 18:53:10 : [INFO ] LISAv2 exit code: 0
PS C:\Users\juhlee\Downloads\LISAv2-juhle-issue652\LISAv2-juhle-issue652>

**Comment by juhlee-microsoft** [comment]
This is the test result from Jenkins.

image

**Comment by juhlee-microsoft** [comment]
@ader1990  @chvalean  Thank you for the comment and feedback. Here I fixed the code and let me know if you have any other question.

**Comment by chvalean** [comment]
IMO the 'Not a git repo' message should not be displayed at all if that's the case.
Please also add a check for the git command if available, it should not be assumed to just work.

**Comment by juhlee-microsoft** [comment]
> IMO the 'Not a git repo' message should not be displayed at all if that's the case.
> Please also add a check for the git command if available, it should not be assumed to just work.

@chvalean I don't agree with your feedback because when it falls into exceptional case which means 'it is not git repo', it should shows branch is from git or not. Also The purpose of this display is where my test result comes from in the report.  'Not git repo' message does not harm in the report.

**Comment by juhlee-microsoft** [comment]
@chvalean @ader1990 @LiliDeng @lizzha I removed two tags and make this ready to merge to the master. Sign off or leave the comment by today.

File: LISAv2-Framework.psm1 [hunk 4]
```diff
@@ -78,6 +78,26 @@ function Start-LISAv2 {
...
+						Write-Debug "Found branch information and updated with $current_branch"
+					}
+				}
+				$CommitId = git log --format="%ai`t%H`t%an`t%ae`t%s" -n 1 | ConvertFrom-String -Delimiter "`t"
+				$CommitId = $CommitId.P2
+			} catch {
+				Write-LogWarn "This may not be git repo. Can not find branch and commit id, but ok for the rest of processing"
+			} finally {
+				$current_branch = "Not git repo"
+				$CommitId = "0"
```
Comment (lizzha) [inline LISAv2-Framework.psm1]: these two lines in the finally block overwrites the value of the variables, how can it output the branch and commit ID later?

File: LISAv2-Framework.psm1 [hunk 5]
```diff
@@ -78,6 +78,26 @@ function Start-LISAv2 {
 			Set-Variable -Name "TestId" -Value $testId -Scope Global -Force
 			Write-LogInfo "Autogenerated test ID: $testId"
 
+			# Get git information
+			# Set initial value
+			$current_branch = "Not a git repo"
+			$CommitId = "0"
```
Comment (lizzha) [inline LISAv2-Framework.psm1]: initializing the variables is not necessary?

**Comment by santoshx** [comment]
You may consider using 'git describe' for human readable version information.

**Comment by juhlee-microsoft** [comment]
<cmdline test inside git repo>
[LISAv2 Test Results Summary]
Test Run On           : 01/30/2019 08:42:03
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:5

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 2.91
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS


Logs can be found at C:\LISAv2\OA31\TestResults\2019-30-01-00-42-00-3897


01/30/2019 08:47:22 : [INFO ] ----------------------------------------------------------
01/30/2019 08:47:22 : [INFO ] Current Branch:  juhle/issue652
01/30/2019 08:47:22 : [INFO ] Currnet Commit ID: 48b46c535aab039cb445c668661a0ab18263c8df
01/30/2019 08:47:22 : [INFO ] ----------------------------------------------------------
01/30/2019 08:47:22 : [INFO ] Creating 'C:\LISAv2\OA31\Azure-OA31-TestLogs.zip' from 'C:\LISAv2\OA31\TestResults\2019-30-01-00-42-00-3897'
01/30/2019 08:47:22 : [INFO ] C:\LISAv2\OA31\Azure-OA31-TestLogs.zip created successfully.
01/30/2019 08:47:22 : [INFO ] Analyzing results..
01/30/2019 08:47:22 : [INFO ] PASS  : 1
01/30/2019 08:47:22 : [INFO ] FAIL  : 0
01/30/2019 08:47:22 : [INFO ] ABORT : 0
01/30/2019 08:47:22 : [INFO ] Copying all files back to original working directory: C:\Users\juhlee\source\repo\LISAv2.
01/30/2019 08:47:26 : [INFO ] Cleaning up C:\LISAv2\OA31
01/30/2019 08:47:26 : [INFO ] Setting workspace back to original location: C:\Users\juhlee\source\repo\LISAv2
01/30/2019 08:47:26 : [INFO ] LISAv2 exit code: 0
PS C:\Users\juhlee\source\repo\LISAv2>

<pipeline jenkins>
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] [LISAv2 Test Results Summary]
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] Test Run On           : 01/30/2019 08:44:17
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] Total Time (dd:hh:mm) : 0:0:5
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2]    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] ---------------------------------------------------------------------------------------------------------------------
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2]     1 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 2.75 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 	FirstBoot : PASS 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 	FirstBoot : Call Trace Verification : PASS 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 	Reboot : PASS 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 	Reboot : Call Trace Verification : PASS 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] Logs can be found at C:\LISAv2\DO93\TestResults\2019-30-01-08-44-12-4015
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] ------------------------------------------------------------
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] Current Branch: Not a git repo
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] Current Commit ID: 
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] ------------------------------------------------------------
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] Creating 'C:\LISAv2\DO93\Azure-DO93-TestLogs.zip' from 'C:\LISAv2\DO93\TestResults\2019-30-01-08-44-12-4015'
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] C:\LISAv2\DO93\Azure-DO93-TestLogs.zip created successfully.
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] Analyzing results..
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] PASS  : 1
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] FAIL  : 0
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] ABORT : 0
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:39 : [INFO ] Copying all files back to original working directory: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@2.
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:42 : [INFO ] Cleaning up C:\LISAv2\DO93
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:42 : [INFO ] Setting workspace back to original location: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@2
[Azure BVT-VERIFY-DEPLOYMENT-PROVISION westus2] 01/30/2019 08:49:42 : [INFO ] LISAv2 exit code: 0


<cmdline in non-git repo>
[LISAv2 Test Results Summary]
Test Run On           : 01/30/2019 08:49:34
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:6

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 3.76
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS


Logs can be found at C:\LISAv2\LN69\TestResults\2019-30-01-00-49-31-0442


01/30/2019 08:55:40 : [INFO ] ------------------------------------------------------------
01/30/2019 08:55:40 : [INFO ] Current Branch: Not git repo
01/30/2019 08:55:40 : [INFO ] Current Commit ID: 0
01/30/2019 08:55:40 : [INFO ] ------------------------------------------------------------
01/30/2019 08:55:40 : [INFO ] Creating 'C:\LISAv2\LN69\Azure-LN69-TestLogs.zip' from 'C:\LISAv2\LN69\TestResults\2019-30-01-00-49-31-0442'
01/30/2019 08:55:40 : [INFO ] C:\LISAv2\LN69\Azure-LN69-TestLogs.zip created successfully.
01/30/2019 08:55:40 : [INFO ] Analyzing results..
01/30/2019 08:55:40 : [INFO ] PASS  : 1
01/30/2019 08:55:40 : [INFO ] FAIL  : 0
01/30/2019 08:55:40 : [INFO ] ABORT : 0
01/30/2019 08:55:40 : [INFO ] Copying all files back to original working directory: C:\Users\juhlee\Downloads\LISAv2-juhle-issue652\LISAv2-juhle-issue652.
01/30/2019 08:55:44 : [INFO ] Cleaning up C:\LISAv2\LN69
01/30/2019 08:55:44 : [INFO ] Setting workspace back to original location: C:\Users\juhlee\Downloads\LISAv2-juhle-issue652\LISAv2-juhle-issue652
01/30/2019 08:55:44 : [INFO ] LISAv2 exit code: 0
PS C:\Users\juhlee\Downloads\LISAv2-juhle-issue652\LISAv2-juhle-issue652>

File: LISAv2-Framework.psm1 (hunk 5 above)
Comment (juhlee-microsoft) [reply LISAv2-Framework.psm1]: I'll just keep it in case.

File: LISAv2-Framework.psm1 [hunk 6]
```diff
@@ -78,6 +78,25 @@ function Start-LISAv2 {
 			Set-Variable -Name "TestId" -Value $testId -Scope Global -Force
 			Write-LogInfo "Autogenerated test ID: $testId"
 
+			# Get git information
+			# Set initial value
+			$current_branch = "Not a git repo"
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: camelCase for variables

Comment (ader1990) [reply LISAv2-Framework.psm1]: I would initialize the variables with $null, and at log time, print the values if they are not null.

File: LISAv2-Framework.psm1 [hunk 7]
```diff
@@ -78,6 +78,25 @@ function Start-LISAv2 {
...
+			$CommitId = "0"
+			try {
+				git branch | foreach {
+					if($_ -match "^\*(.*)"){
+						$current_branch = $matches[1]
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: camelCase

File: LISAv2-Framework.psm1 [hunk 8]
```diff
@@ -78,6 +78,25 @@ function Start-LISAv2 {
...
+						Write-Debug "Found branch information and updated with $current_branch"
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: don t use write-debug, use lisav2 logging framework

File: LISAv2-Framework.psm1 [hunk 9]
```diff
@@ -78,6 +78,25 @@ function Start-LISAv2 {
 			Set-Variable -Name "TestId" -Value $testId -Scope Global -Force
 			Write-LogInfo "Autogenerated test ID: $testId"
 
+			# Get git information
+			# Set initial value
+			$current_branch = "Not a git repo"
+			$CommitId = "0"
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: commitId

File: LISAv2-Framework.psm1 (hunk 7 above)
Comment (ader1990) [reply LISAv2-Framework.psm1]: currentBranch

File: LISAv2-Framework.psm1 [hunk 10]
```diff
@@ -78,6 +78,25 @@ function Start-LISAv2 {
...
+				git branch | foreach {
+					if($_ -match "^\*(.*)"){
+						$current_branch = $matches[1]
+						Write-Debug "Found branch information and updated with $current_branch"
+					}
+				}
+				$CommitId = git log --format="%ai`t%H`t%an`t%ae`t%s" -n 1 | ConvertFrom-String -Delimiter "`t"
+				$CommitId = $CommitId.P2
+			} catch {
+				Write-LogWarn "This may not be git repo. Can not find branch and commit id, but ok for the rest of processing"
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: you don t need to do nothing on catch/finally. it does not add any value in my opinion to print a message if this is not a git repo or if you do not have a git binary in the path.

File: LISAv2-Framework.psm1 [hunk 11]
```diff
@@ -167,6 +186,11 @@ function Start-LISAv2 {
 			$plainTextSummary = $testController.TestSummary.GetPlainTextSummary()
 			Write-LogInfo $plainTextSummary
 
+			Write-LogInfo "------------------------------------------------------------"
```
Comment (ader1990) [inline LISAv2-Framework.psm1]: print this only if there is a currentBranch and commitId. And print it on one liner, there is no need to put more info than needed.

**Pull request closed without merging**